AppCenterData/AppCenterData/Internal/Util/MSDocumentUtils.m

```diff
@@ -163,15 +163,18 @@ + (MSDocumentWrapper *)documentWrapperFromDictionary:(NSDictionary *)dictionary
     deserializedValue = [(id<MSSerializableDocument>)[documentType alloc] initFromDictionary:dictionary];
     MSLogDebug([MSData logTag], @"Successfully deserialized document: %@ (partition: %@)", documentId, partition);
   }
-  return [[MSDocumentWrapper alloc] initWithDeserializedValue:deserializedValue
-                                                    jsonValue:jsonValue
-                                                    partition:partition
-                                                   documentId:documentId
-                                                         eTag:eTag
-                                              lastUpdatedDate:lastUpdatedDate
-                                             pendingOperation:pendingOperation
-                                                        error:dataError
-                                              fromDeviceCache:fromDeviceCache];
+  if (dataError) {
+    return [[MSDocumentWrapper alloc] initWithError:dataError documentId:documentId partition:partition eTag:eTag];
```

Comment:
Let's not drop `lastUpdatedDate`, `pendingOperation` and `fromDeviceCache` in the `MSDocumentWrapper` object returned here. Let's extends the initializer of `initWithError:` to take those 3 parameters.

Reply:
@dhei I've added another overload to take those parameters. That could change based on @achocron 's feedback.